test/integration/hds_integration_test.cc

```diff
@@ -75,56 +67,145 @@ class HdsIntegrationTest : public HttpIntegrationTest,
   }
 
   void cleanupHdsConnection() {
-    if (fake_hds_connection_ != nullptr) {
-      fake_hds_connection_->close();
-      fake_hds_connection_->waitForDisconnect();
+    if (hds_fake_connection_ != nullptr) {
+      hds_fake_connection_->close();
+      hds_fake_connection_->waitForDisconnect();
     }
   }
 
-  static constexpr uint32_t upstream_endpoints_ = 5;
+  envoy::service::discovery::v2::HealthCheckSpecifier makeHealthCheckSpecifier() {
+    // HealthCheckSpecifier
+    envoy::service::discovery::v2::HealthCheckSpecifier server_health_check_specifier;
+    server_health_check_specifier.mutable_interval()->set_nanos(500000000); // 500ms
+    auto* endpoint_address = server_health_check_specifier.add_health_check()
+                                 ->add_endpoints()
+                                 ->add_endpoints()
+                                 ->mutable_address()
+                                 ->mutable_socket_address();
+    endpoint_address->set_address(Network::Test::getLoopbackAddressString(version_));
+    endpoint_address->set_port_value(host_upstream_->localAddress()->ip()->port());
+
+    auto* anna = server_health_check_specifier.mutable_health_check(0)->mutable_endpoints(0);
+    anna->mutable_locality()->set_region("some_region");
+    anna->mutable_locality()->set_zone("some_zone");
+    anna->mutable_locality()->set_sub_zone("crete");
+
+    auto* cluster_health_check =
+        server_health_check_specifier.mutable_health_check(0)->add_health_checks();
+    cluster_health_check->mutable_timeout()->set_seconds(1);
+    cluster_health_check->mutable_interval()->set_seconds(1);
+    cluster_health_check->mutable_unhealthy_threshold()->set_value(2);
+    cluster_health_check->mutable_healthy_threshold()->set_value(2);
+    cluster_health_check->mutable_grpc_health_check();
+    auto* http_health_check = cluster_health_check->mutable_http_health_check();
+    http_health_check->set_use_http2(false);
+    http_health_check->set_path("/healthcheck");
+
+    return server_health_check_specifier;
+  }
+
+  static constexpr uint32_t upstream_endpoints_ = 0;
 
   IntegrationStreamDecoderPtr response_;
   std::string sub_zone_{"winter"};
-  FakeHttpConnectionPtr fake_hds_connection_;
+  FakeHttpConnectionPtr hds_fake_connection_;
   FakeStreamPtr hds_stream_;
   FakeUpstream* hds_upstream_{};
+  FakeUpstream* host_upstream_{};
```

Comment:
This should probably be a `FakeUpstreamPtr` to deal with cleanup.

Reply:
There is no FakeUpstreamPtr.